addons/sale_loyalty/models/sale_order.py

```diff
@@ -634,18 +713,25 @@ def _get_claimable_rewards(self, forced_coupons=None):
         self.ensure_one()
         all_coupons = forced_coupons or (self.coupon_point_ids.coupon_id | self.order_line.coupon_id | self.applied_coupon_ids)
         has_payment_reward = any(line.reward_id.program_id.is_payment_program for line in self.order_line)
-        total_is_zero = float_is_zero(self.amount_total, precision_digits=2)
-        result = defaultdict(lambda: self.env['loyalty.reward'])
         global_discount_reward = self._get_applied_global_discount()
+        discountable, _discountable_per_tax = self._discountable_order(global_discount_reward, is_reward_mandatory=False)
+        total_is_zero = float_is_zero(discountable, precision_digits=2)
```

Comment:
Yes, I saw that, but I don't understand how changing self.amount_total to discountable is strictly related to this PR and not a fix that has to be done in previous versions as well.

Reply:
I didn't do this to fix anything but to avoid recalculating `discountable` in `_is_better_global_discount_applied` which is called in a loop. And as `_is_better_global_discount_applied` doesn't take the highest discount but the best one (ex: if my order is 100$ and I have a 100$ coupon and another one of 150$, it will take the 100$ and not the 150$ one), I had to change the `self.amount_total` which in my example would have been 0$ and therefore wouldn't have been able to apply the 100$ coupon if the 150$ one was applied.